124-stretch-cluster.md

```diff
@@ -0,0 +1,1795 @@
+The operator on the central cluster creates all necessary resources for the node pool on the target cluster.
+
+#### Resource Distribution
+
+The following table shows which resources are created in each cluster:
+
+| Resource Type | Central Cluster | Remote Clusters |
+|--------------|----------------|----------------|
+| **Custom Resources (Kafka, KafkaNodePool)** | Created by user | Never created |
+| **StrimziPodSet** | Created for central pools | Created by central operator |
+| **Pods** | Managed by local operator | Managed by local operator |
+| **Services** | Created for central pods | Created by central operator |
+| **ConfigMaps** | Created for central pods | Created by central operator |
+| **Secrets (CA, certs)** | Created once | Replicated by central operator |
+| **PersistentVolumeClaims** | Created for central pools | Created by central operator |
+| **ServiceExport/Routes/Ingress** | Per networking provider | Per networking provider |
+
+The central operator has write access to remote clusters via kubeconfig and creates resources directly.
+Remote operators are responsible only for reconciling StrimziPodSet resources (creating/updating pods).
+
+### Configuration Design Rationale
+
+#### Why Stretch Configuration is at Operator Level
+
+The stretch cluster configuration (`STRIMZI_CENTRAL_CLUSTER_ID`, `STRIMZI_REMOTE_KUBE_CONFIG`, plugin settings) is specified at the cluster operator level rather than at the Kafka CR or KafkaNodePool level.
+This design decision addresses several important architectural and operational concerns.
+
+##### Infrastructure-Level Concern
+
+Stretch cluster functionality requires infrastructure-level connectivity and authentication to remote Kubernetes clusters.
+The operator needs kubeconfig credentials and API server endpoints to create resources in remote clusters.
+These credentials grant cluster wide access to Kubernetes APIs and are managed by platform administrators, not application developers.
+
+Placing these credentials at the operator level:
+
+- Aligns with Kubernetes security model where operators run with ServiceAccounts that have cluster-level permissions
+- Avoids embedding sensitive kubeconfig credentials in application level CRs (Kafka, KafkaNodePool).
+- Simplifies credential rotation by updating operator deployment once rather than every Kafka CR
+- Prevents credential sprawl across multiple Kafka instances
+
+##### Single Networking Provider Per Operator
+
+The networking plugin (NodePort, LoadBalancer, MCS) is loaded once at operator startup via Java ServiceLoader.
+The plugin JAR is mounted as a volume in the operator deployment.
+The plugin's initialization code runs once when the operator starts.
+
+This means all Kafka clusters managed by a single operator instance must use the same networking provider.
+Different networking providers cannot coexist in a single operator instance.
+If networking provider were configurable per Kafka CR, the operator would need to:
```

Comment:
I think the proposal could be simplified by moving what wasn't picked as solution in the rejected alternatives section. The proposal should be just what you proposed and how it should work in your opinion. All the reasoning about other not-picked choices should not be there. Reviewers can go through the rejected alternatives to read about them. I think this is happening in more places across the proposal that could be simplified at the core this way.